src/ArchMap.cpp

```diff
@@ -149,7 +149,22 @@ std::string SleighIdFromCore(RCore *core)
 {
 	const char *arch = r_config_get(core->config, "asm.arch");
 	auto arch_it = arch_map.find(arch);
-	if(arch_it == arch_map.end())
-		throw LowlevelError("Could not match asm.arch " + std::string(arch) + " to sleigh arch.");
+	if(arch_it == arch_map.end()) {
+		char *cpu = strdup (r_config_get(core->config, "asm.cpu"));
+		char *colon = cpu;
+		while (*colon) {
+			if (*colon == ':') {
+				*colon = 0;
+				break;
+			}
+			*colon = tolower (*colon);
+			colon++;
+		}
+		arch_it = arch_map.find(cpu);
```

Comment:
This is wrong, `cpu` is the sleigh id while `arch_map` is keyed on r2 plugin names, so they are unrelated.

Reply:
i know. thats why im splitting the asm.cpu string to the first colon which basically gives me the arch name. and this thing maps fine with the r2 plugin. otherwise its impossible to use the decompiler when using the ghidra disassemblers. which solution do you propose here?

Reply:
> and this thing maps fine with the r2 plugin.

If it does, it is only by coincidence.

> which solution do you propose here?

Return the `asm.cpu` value as-is. It is already the full sleigh id you want.